pkg/services/ngalert/state/state.go

```diff
@@ -124,3 +124,18 @@ func (a *State) Equals(b *State) bool {
 		a.LastEvaluationTime == b.LastEvaluationTime &&
 		data.Labels(a.Annotations).String() == data.Labels(b.Annotations).String()
 }
+
+func (a *State) TrimResults(alertRule *ngModels.AlertRule) {
+	numBuckets := 2 * (int64(alertRule.For.Seconds()) / alertRule.IntervalSeconds)
```

Comment:
Can the IntervalSeconds be zero? This will panic if so.

Reply:
We enforce the minimum of 10s in the UI (as of a fresh PR from @domasx2 ) and in the code where we create or update the rule config.